userspace/sysdig/plugin_utils.cpp

```diff
@@ -155,39 +163,146 @@ void select_plugin(string& name, const string& init_config)
 
 			if (fname == name || fname == soname)
 			{
-				g_selected_plugins.emplace(name, plugin_selected_info{fpath, init_config});
+				g_selected_plugins_init.emplace(name, plugin_selected_init{fpath, init_config});
 				found = true;
 				break;
 			}
 
 			tinydir_next(&dir);
 		}
 
-		tinydir_close(&dir);
+		if (!found)
+		{
+			tinydir_close(&dir);
+		}
+		else
+		{
+			break;
+		}
 	}
+
 	if (!found)
 	{
 		throw sinsp_exception("plugin " + name + " not found. Use -Il to list all installed plugins.");
 	}
 }
 
-shared_ptr<sinsp_plugin> enable_plugin(sinsp *inspector, string& name)
+void select_plugin_enable(string& name, const string& open_params)
 {
-	auto itr = g_selected_plugins.find(name);
-	if (itr == g_selected_plugins.end())
-	{
-		throw sinsp_exception("plugin " + name + " not loaded. Use -H to load it.");
-	}
-	auto plugin = sinsp_plugin::register_plugin(inspector, itr->second.path, itr->second.init_config.c_str());
-	return plugin;
+    g_selected_plugins_enable.emplace(name, open_params);
```

Comment:
@LucaGuerra btw why are we allowing to enable multiple source plugin just to raise an exception (below in `enable_source_plugin`) if more than one is present?